packages/caliper-ethereum/lib/ethereum.js

```diff
@@ -161,25 +171,32 @@ class Ethereum extends BlockchainInterface {
      */
     async sendTransaction(context, contractID, contractVer, methodCall, timeout) {
         let status = new TxStatus();
+        let params = {from: context.fromAddress}
         try {
             context.engine.submitCallback(1);
             let receipt = null;
             let methodType = 'send';
             if (methodCall.isView) {
                 methodType = 'call';
+            } else {
+                let nonce = context.nonces[context.fromAddress]
+                context.nonces[context.fromAddress] = nonce + 1
```

Comment:
`getTransactionCount` won't work because it will not consider all the transactions in the memory pool, just what is in the latest block or what is in the block ready to be mined. There is almost always some lag between the tx being received and the block being ready to mine. 
 This is especially tough in multi-node environments where the transactions may not have already propagated and there may be gaps in the nonces.  

Perhaps we should move the nonce cache to a global context?

Reply:
The problem is the "global" part. The clients run in different processes (or even different nodes once the distributed client feature is reimplemented). Syncing this value after every TX an account makes is a big overhead.
A viable solution would be to distribute the usable accounts among the clients using the `prepareClients` adapter function, and the adapter could distribute the addresses in a round-robin fashion. If you only have one address, then only one client can use it (like now). 
I don't know whether the adapter supports multiple sender addresses (I think not), so it's another todo.

Reply:
Multiple addresses is also a good long term evolution as the block production and gossiping can be easier when the same address isn't used over and over.  One address per process would be the simplest solution.